### ISSUE TITLE
<AnimatePresence initial={false} /> does not work as expected

### ISSUE THREAD

**Issue by marklawlor** [nandorojo/moti#55]
## Issue

This code will play the initial animation, even while initial is set to false

```
<AnimatePresence initial={false}>
 <View
      from={{
        opacity: 0,
        scale: 0.9,
      }}
      animate={{
        opacity: 1,
        scale: 1,
      }}
      exit={{
        opacity: 0,
        scale: 0.9,
      }}
      style={styles.shape}
    />
</AnimatePresence>
```

## Workaround

Rename the `from` prop to `initial`

```
<AnimatePresence initial={false}>
 <View
      initial={{
        opacity: 0,
        scale: 0.9,
      }}
      animate={{
        opacity: 1,
        scale: 1,
      }}
      exit={{
        opacity: 0,
        scale: 0.9,
      }}
      style={styles.shape}
    />
</AnimatePresence>
```

I'm not sure if this is expected behaviour as the initial prop is not documented and how it differs to the `from` prop

**Comment by nandorojo** [comment]
As it stands, animate presence only supports exits for presence animations. I'm not exactly sure how framer implements the `initial` prop, but it isn't used on Moti at the moment. What is the expectation of what should happen here? Should the initial state be `animate`?

**Comment by marklawlor** [comment]
I believe Framers `initial` prop is basically the same as moti's `from` prop. I think if the AnimatePresence cloned its children and copied the `from` prop to `initial` there's a chance it might just work.

**Comment by nandorojo** [comment]
I don't think so - Moti leverages a headless version of AnimatePresence with the `usePresence` hook, and implements the exit style when `!isPresent`, and then calls `safeToUnmount` when the final reanimated callback fires.

So my guess is we'd need to make a custom context that checks for the `initial` value and export a custom `AnimatePresence component.`

**Comment by nandorojo** [comment]
Reading through this code: https://github.com/framer/motion/blob/main/src/components/AnimatePresence/PresenceChild.tsx

It looks like we could just get the `initial` value from the presence child context.

**Comment by Stevemoretz** [comment]
I have the same issue

**Comment by Stevemoretz** [comment]
Providing an empty object to from on the first render and then useEffect to set from to what should really be, does fix the issue, as a workaround. ;)

**Comment by nandorojo** [comment]
Could you please try this https://github.com/nandorojo/moti/pull/102

```sh
yarn add moti@presence
```

I added support for the `initial` prop there. Would be useful to get it tested.

**Comment by nandorojo** [comment]
Fixed in `0.12.0`

```sh
yarn add moti
```
